### ISSUE TITLE
Add support for the Elastic Common Schema for existing parsers

### ISSUE THREAD

**Issue by archoversight** [vectordotdev/vector#10836]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Current Vector Version

```
latest
```

### Use-cases

Vector has a variety of parsers that automatically take input from various sources and formats it into JSON, it would be great if these were extended to support mapping to [ECS (Elastic Common Schema)](https://www.elastic.co/guide/en/ecs/current/index.html) since that will allow the data to be easily ingested by and viewed in Elastic/Kibana.

### Attempted Solutions

Currently we can use VRL to write mappings for each of the different formats, as well as the enrichment of common fields used, or we can use Elastic Search's pipelines/indexing to do the remapping.

If however Vector outputs data in ECS by default it means there's one common place to find definitions and the schema for the outputted data vs needing to understand each of the different pieces of data that exist right now.

### Proposal

No proposal, other than this would be neat to have implemented by default to reduce the need for manually implementing this functionality in VRL or in pipelines.

### References

- https://www.elastic.co/guide/en/ecs/current/index.html
- https://www.elastic.co/blog/easily-ingest-data-to-elastic-via-splunk

**Comment by spencergilbert** [comment]
Duplicate of: https://github.com/vectordotdev/vector/issues/2423

We've begun adding support for schemas already, but started with a much smaller scope. Once that lands (soon), we'll be looking to add support for things like ECS as well as OTEL 👍
